src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/Deliveries.java

```diff
@@ -94,6 +93,32 @@ public boolean lastDeliveryIsApprovedOrDisapproved(Assignment assignment, Group
             .sorted(byState.thenComparing(bySubmissionDate))
             .collect(Collectors.toList());
     }
+    
+    /**
+     * Get the most recent delivery for every group in this assignment
+     * @param assignment current assignment
+     * @return a list of deliveries
+     */
+    @Transactional
+    public List<Delivery> getLastDeliveries(Assignment assignment) {
+    	return this.getLastDeliveries(assignment, null);
+    }
+    
+    /**
+     * Get the most recent delivery for every group in this assignment with the specific state
+     * @param assignment current assignment
+     * @return a list of deliveries
+     */
+    @Transactional
+    public List<Delivery> getLastDeliveries(Assignment assignment, String state) {
+    	BooleanExpression expression = QDelivery.delivery.assignment.eq(assignment);
+
+    	if (state != null) {
```

Comment:
But this would return all the submissions, but then in a map? Personally I think a 'and' construct is better than a map of all deliveries sorted, but I can understand the reasoning for choosing the latter. If you want this, I can change it of course.